src/MatrixFactorizations.jl

```diff
@@ -67,6 +67,7 @@ function _mul(Q::LayoutQ, b::AbstractVector)
    lmul!(convert(AbstractQtype{T}, Q), bnew)
 end
 (*)(Q::LayoutQ, B::AbstractMatrix) = _mul(Q, B)
+(*)(Q::LayoutQ, B::LayoutQ) = ArrayLayouts.mul(Q, B)
```

Comment:
@dlfivefifty this line is repeated in 124 which breaks precompilation in pre 1.10 https://github.com/SciML/NonlinearSolve.jl/issues/402

https://github.com/SKopecz/PositiveIntegrators.jl/actions/runs/8597851891/job/23557395790?pr=61#step:6:215